tests/test_observation.py

```diff
@@ -30,82 +30,85 @@ def get_psfs(self, shape, sigmas):
 
         skycoord = [210.945, -73.1]
-        assert_array_equal(scene.get_pixel(skycoord), [-110, -202])
+        assert_array_equal(frame.get_pixel(skycoord), [-110, -202])
 
     def test_init(self):
         images = np.arange(1, 430).reshape(3, 11, 13)
         weights = np.ones_like(images)
         psfs = np.arange(1, 76).reshape(3, 5, 5)
-        norm_psfs = psfs / np.array([325, 950, 1575])[:, None, None]
+        norm_psfs = psfs / psfs.sum(axis=(1,2))[:, None, None]
         wcs = get_airy_wcs()
-        structure = (0, 1, 1, 1, 0)
+        bands = np.arange(len(images))
 
         # Minimal init
         obs = scarlet.Observation(images)
-        assert obs.B == 3
-        assert obs.Ny == 11
-        assert obs.Nx == 13
-        assert obs.shape == images.shape
-        assert obs.psfs is None
-        assert_array_equal(obs.get_pixel((5.1, 1.3)), (5, 1))
+        assert obs.frame.B == 3
+        assert obs.frame.Ny == 11
+        assert obs.frame.Nx == 13
+        assert obs.frame.shape == images.shape
+        assert obs.frame.psfs is None
+        assert_array_equal(obs.frame.get_pixel((5.1, 1.3)), (5, 1))
         assert obs.weights == 1
-        assert obs.structure is None
+        assert obs.frame.bands is None
 
         # Full init
-        obs = scarlet.Observation(images, psfs, weights, wcs, structure=structure)
-        assert obs.B == 3
-        assert obs.Ny == 11
-        assert obs.Nx == 13
-        assert obs.shape == images.shape
-        assert_almost_equal(obs.psfs, norm_psfs)
-        assert_almost_equal(obs.psfs.sum(axis=(1, 2)), [1]*3)
+        obs = scarlet.Observation(images, psfs=norm_psfs, weights=weights, wcs=wcs, bands=bands)
+        assert obs.frame.B == 3
+        assert obs.frame.Ny == 11
+        assert obs.frame.Nx == 13
+        assert obs.frame.shape == images.shape
+        assert_almost_equal(obs.frame.psfs, norm_psfs)
+        assert_almost_equal(obs.frame.psfs.sum(axis=(1, 2)), [1]*3)
         assert_array_equal(obs.weights, weights)
-        assert_array_equal(obs.structure, structure)
+        assert_array_equal(obs.frame.bands, bands)
 
         skycoord = [210.945, -73.1]
-        assert_array_equal(obs.get_pixel(skycoord), [-110, -202])
+        assert_array_equal(obs.frame.get_pixel(skycoord), [-110, -202])
 
     def test_psf_match(self):
         shape = (43, 43)
         target_psf = self.get_psfs(shape, [.9])[1][0]
+        target_psf = target_psf[None]
         psfs, truth = self.get_psfs(shape, [2.1, 1.1, 3.5])
+        psfs /= psfs.sum(axis=(1,2))[:,None,None]
 
-        scene = scarlet.Scene(psfs.shape, psfs=target_psf)
+        frame = scarlet.Frame(psfs.shape, psfs=target_psf)
         observation = scarlet.Observation(psfs, psfs)
-        observation.match(scene)
-        result = observation.get_model(np.array([target_psf]*len(psfs)))
+        observation.match(frame)
+        result = observation.render(np.array([target_psf]*len(psfs)))
 
```

Comment:
Ok, I figured out why this test is failing. You added line 95, which changes the target PSF shape to go from (Ny, Nx) to (1, Ny, Nx), so when they are compared in the assert statement they have different lengths. This is because one of the things this was testing is that `Scene` assumes that a 2D input PSF is a single band PSF and adds the extra dimension, which the new version of the code no longer does. Rather than make a change myself that might cause a conflict in your branch, I'll give you the updated code below:

```
def test_psf_match(self):
    shape = (43, 43)
    target_psf = self.get_psfs(shape, [.9])[1]
    psfs, truth = self.get_psfs(shape, [2.1, 1.1, 3.5])
    psfs /= psfs.sum(axis=(1,2))[:,None,None]

    frame = scarlet.Frame(psfs.shape, psfs=target_psf)
    observation = scarlet.Observation(psfs, psfs)
    observation.match(frame)
    result = observation.render(np.array([target_psf[0]]*len(psfs)))

    assert_almost_equal(result, truth)
```